mods/mc_tutorial/init.lua

```diff
@@ -0,0 +1,1708 @@
+            context.reward_selected = {
+                [RW_LIST] = new_index,
+                [RW_SELECT] = math.max(1, math.min(context.reward_selected[RW_SELECT], #context.selected_rewards)),
+                ["active"] = RW_LIST
+            }
+            reload = true
+        end
+        if fields.reward_search_go then
+            -- TODO
+        end
+        if fields.reward_search_x then
+            -- TODO
+        end
+
+        -- EVENTS TAB INTERACTIONS
+        local eventlist_field_active = false
+        for k,v in pairs(fields) do
+            if string.sub(k, 1, 10) == "eventlist_" then
+                eventlist_field_active = true
+                break
+            end
+        end
+        if eventlist_field_active then
+            reload = true
+            if fields.eventlist_add_event then
+                context.selected_event = context.selected_event or 1
+                return mc_tutorial.show_event_popop_fs(player, false)
+            end
+            if fields.eventlist_add_group then
+                context.selected_event = context.selected_event or 1
+                if not mc_tutorial.record.temp[pname].sequence[context.selected_event] then
+                    minetest.chat_send_player(pname, "[Tutorial] Groups can not be added to empty tutorials.")
+                elseif mc_tutorial.record.temp[pname].sequence[context.selected_event].action == mc_tutorial.ACTION.GROUP then
+                    minetest.chat_send_player(pname, "[Tutorial] Groups can not be added around group markers.")
+                elseif group_parity_is_odd(context.selected_event, mc_tutorial.record.temp[pname].sequence) then
+                    minetest.chat_send_player(pname, "[Tutorial] Groups can not be added inside other groups.")
+                else
+                    local group = mc_tutorial.record.temp[pname].next_group or 1
+                    table.insert(mc_tutorial.record.temp[pname].sequence, context.selected_event + 1, {
+                        action = mc_tutorial.ACTION.GROUP,
+                        g_type = mc_tutorial.GROUP.END,
+                        g_id = group,
+                        g_remaining = {},
+                        g_length = 0,
+                    })
+                    table.insert(context.events, context.selected_event + 1, "#CCFFFF} END GROUP "..group)
+                    table.insert(mc_tutorial.record.temp[pname].sequence, context.selected_event, {
+                        action = mc_tutorial.ACTION.GROUP,
+                        g_type = mc_tutorial.GROUP.START,
+                        g_id = group,
+                    })
+                    table.insert(context.events, context.selected_event, "#CCFFFFGROUP "..group.. " {")
+
+                    mc_tutorial.record.temp[pname].next_group = group + 1
```

Comment:
This leads to sequential group numbering even after the earlier groups are deleted. For example, I can create and delete 3 groups, then create another group and the index starts at GROUP 4 instead of 1.

Reply:
Noted; this is being done to ensure that no groups have internal IDs that overlap. Can look into changing the way groups are numbered so that group IDs automatically update as groups are removed

Reply:
Existing group IDs are now automatically re-indexed when groups are deleted